server/src/messageHandler.mjs

```diff
@@ -80,18 +83,47 @@ async function handleMessage(message, userId, ws) {
   }
 
   // Handle JSON-RPC messages that are event listener enable requests
-
-  if ( events.isEventListenerOnMessage(oMsg) ) {
-    events.sendEventListenerAck(userId, ws, oMsg);
-    events.registerEventListener(userId, oMsg, ws);
-    return;
-  }
-
+  // First we extract event data from the message using the registrationMessage config
+  // Then we register the event listener for the specified user and WebSocket with the extracted metadata
+  if (events.isEventListenerOnMessage(oMsg)) {
+    const eventMetadata = events.extractEventData(oMsg, eventConfig.registrationMessage, true);
+  
+    // Only send ack messages and return if not in proxy mode
+    if (!process.env.proxy) {
+      // If registrationAck config is included, send ack message
+      if (eventConfig.registrationAck) {
+        events.sendEventListenerAck(userId, ws, eventMetadata);
+      }
+  
+      events.registerEventListener(userId, eventMetadata, ws);
+      return;
+    }
+    // If in proxy mode, just register the event listener without sending ack messages or returning
+    else {
+      events.registerEventListener(userId, eventMetadata, ws);
```

Comment:
There's some duplicate code here. We always have to register the listener so we can do that first without putting it in the if/else block.